sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -325,6 +326,61 @@ object LogicalPlanIntegrity {
       LogicalPlanIntegrity.hasUniqueExprIdsForOutput(plan))
   }
 
+  /**
+   * This method validates there are no dangling attribute references.
+   * Returns an error message if the check does not pass, or None if it does pass.
+   */
+  def validateNoDanglingReferences(plan: LogicalPlan): Option[String] = {
+    plan.collectFirst {
+      // DML commands and multi instance relations (like InMemoryRelation caches)
+      // have different output semantics than typical queries.
```

Comment:
Do we really need to special-case these two? `QueryPlan#missingInputs` should have already taken care of them
```
  final def missingInput: AttributeSet = references -- inputSet
```

Reply:
Anyway, not a big deal, it's faster to skip some cases that will never hit missing attr issue.